File: src/main/java/net/tlalka/imager/view/SysReport.java
package net.tlalka.imager.view;

import net.tlalka.imager.data.GeoImage;

public class SysReport implements ReportApi {

    private static final String HEADER = "\n\nDIRECTORY: %s\n";
    private static final String FORMAT = "\n%s\t\t\tV(dist): %8.2f\t\t V(cord): %6.0f  ( %s )";
    private static final String FOOTER = "\n\nDONE.\n";

    @Override
    public void header(String directory) {
        print(HEADER, directory);
    }

    @Override
    public void write(GeoImage image) {
        print(FORMAT, image.getFile().getName(), image.getDistance(), image.getDirection(), image.getCardinal());
    }

    @Override
    public void save() {
        print(FOOTER);
    }

    private void print(String message, Object... objects) {
        System.out.printf(message, objects);
    }
}

File: src/main/kotlin/net/tlalka/imager/view/SysReport.kt
package net.tlalka.imager.view

import net.tlalka.imager.data.GeoImage

class SysReport : ReportApi {

    override fun header(header: String) {
        print(HEADER, header)
    }

    override fun write(image: GeoImage) {
        print(FORMAT, image.file.name, image.distance, image.direction, image.cardinal)
    }

    override fun save() {
        print(FOOTER)
    }

    private fun print(message: String, vararg objects: Any) {
        System.out.printf(message, *objects)
    }

    companion object {

        private val HEADER = "\n\nDIRECTORY: %s\n"
        private val FORMAT = "\n%s\t\t\tV(dist): %8.2f\t\t V(cord): %6.0f  ( %s )"
        private val FOOTER = "\n\nDONE.\n"
    }
}